test/standalone/x86_cpu_model/build.zig

```diff
@@ -0,0 +1,57 @@
+const std = @import("std");
+
+pub fn build(b: *std.Build) void {
+    const test_step = b.step("test", "Test it");
+    b.default_step = test_step;
+
+    const target = b.standardTargetOptions(.{});
+    const optimize = b.standardOptimizeOption(.{});
+
+    const arch = target.result.cpu.arch;
+    if (!arch.isX86()) return;
+
+    const abi = target.result.abi;
+    if (target.result.ofmt != .elf or !(abi.isMusl() or abi.isGnu())) return;
+
+    const exe = b.addExecutable(.{
+        .name = "main",
+        .optimize = optimize,
+        .target = target,
+    });
```

Comment:
I would suggest adding an extra step here that actually runs the executable. The test program itself should return the value from `foo_impl` as the exit code from `main`. Then you can make the build script expect the appropriate exit code depending on the CPU features of the build host.